pybamm/models/full_battery_models/lithium_ion/dfn.py

```diff
@@ -53,7 +53,10 @@ def __init__(self, options=None, name="Doyle-Fuller-Newman model", build=True):
 
     def set_porosity_submodel(self):
 
-        self.submodels["porosity"] = pybamm.porosity.Constant(self.param)
+        if self.options["porosity"] is None:
+            self.submodels["porosity"] = pybamm.porosity.Constant(self.param)
+        elif self.options["porosity"] == "variable porosity":
+            self.submodels["porosity"] = pybamm.porosity.Full(self.param)
```

Comment:
equivalent changes need to be made in `SPM` and `SPMe` models

Reply:
Done.